helix-core/src/main/java/org/apache/helix/model/Message.java

```diff
@@ -82,6 +82,7 @@
     RESOURCE_TAG,
     FROM_STATE,
     TO_STATE,
+    TO_STATUS,
```

Comment:
Add a field could be make the message logic more complicated. You can let the from state as "NORMAL" to "FREEZE" or something.

I think you are trying to make an extra field to differentiate whether it is freeze message or not. But I would suggest to use our message type to achieve that goal.